src/main/java/com/endava/web/application/domain/service/EmployeeService.java

```diff
@@ -0,0 +1,80 @@
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class EmployeeService {
+
+    private final EmployeeRepository employeeRepository;
+    private final DepartmentRepository repositoryDepartment;
+
+    public List<Employee> getAllEmployees() {
+        return employeeRepository.findAll();
+    }
+
+    public Employee getEmployee(int employeeId) {
+        Optional<Employee> employee = employeeRepository.findById(employeeId);
+        return employee.orElseThrow(() -> new NoSuchEmployeeException(
+                "EMPLOYEE WITH ID " + employeeId + " DOES NOT EXISTS"));
+    }
+
+    public Employee saveEmployee(Employee employee) {
+        checkEmployeeNameConstraints(employee);
+        if(!repositoryDepartment.findDepartmentById(employee.getDepartment().getId()).isPresent()){
+            throw new NoSuchDepartmentException("DEPARTMENT WITH ID " + employee.getDepartment().getId() + " DOES NOT EXISTS");
+        }
+        return employeeRepository.save(employee);
+    }
+
+    public Employee updateEmployee(Employee employee) {
+        if(!repositoryDepartment.existsById(employee.getDepartment().getId())){
+           throw new DepartmentDoesNotExistsException("DEPARTMENT WITH THIS ID DOES NOT EXISTS");
+        }
+        checkEmployeeNameConstraints(employee);
+        return employeeRepository.save(employee);
+    }
+
+    private void checkEmployeeNameConstraints(Employee employee){
+        if (employee.getFirstName() == null || employee.getFirstName().isEmpty() || employee.getFirstName().trim().isEmpty()) {
+            throw new DepartmentConstraintsException("EMPLOYEE FIRST NAME CANNOT BE NULL, EMPTY OR BLANK");
+        }
+        if (employee.getLastName() == null || employee.getLastName().isEmpty() || employee.getLastName().trim().isEmpty()) {
+            throw new DepartmentConstraintsException("EMPLOYEE LAST NAME CANNOT BE NULL, EMPTY OR BLANK");
+        }
+        if (employee.getEmail() == null || employee.getEmail().isEmpty() || employee.getEmail().trim().isEmpty()) {
+            throw new DepartmentConstraintsException("EMPLOYEE EMAIL CANNOT BE NULL, EMPTY OR BLANK");
+        }
+        if (employee.getPhoneNumber() == null || employee.getPhoneNumber().isEmpty() || employee.getPhoneNumber().trim().isEmpty()) {
+            throw new DepartmentConstraintsException("EMPLOYEE PHONE NUMBER CANNOT BE NULL, EMPTY OR BLANK");
+        }
+        ifEmailAndPhoneExists(employee);
+        if (employee.getSalary() < 1.00) {
+            throw new DepartmentConstraintsException("EMPLOYEE SALARY MUST BE >= 1.0");
+        }
+    }
+
+    private void ifEmailAndPhoneExists(Employee employee) {
+        String email = Boolean.toString(employeeRepository.existsByEmail(employee.getEmail()));
+        String phone = Boolean.toString(employeeRepository.existsByPhoneNumber(employee.getPhoneNumber()));
+        switch(email + "|" + phone) {
```

Comment:
do not transform boolean to string and concat them, better replace with 
if(true && true)
...

Reply:
Done!